wordpress-mobile/WordPress-Android
# Reader Improvements: Sites/Blogs you might like

Project Issue: #12899
Fixes #12834

This is a main branch for the **Reader Improvements: Sites/Blogs you might like** task.
List of related PRs can be found in #12834.

Light mode | Dark mode
------------|------------
![device-2020-10-17-182144](https://user-images.githubusercontent.com/1405144/96337564-c8d54380-10a5-11eb-8bb2-4ce191ed632d.png)|![device-2020-10-17-182125](https://user-images.githubusercontent.com/1405144/96337565-cc68ca80-10a5-11eb-9ab6-1268d69bcae7.png)

**To test**
Bulk of PRs reviewed as part of #12834, needs one more round of testing after [recent recommended blogs API changes](https://github.com/wordpress-mobile/WordPress-Android/issues/13027#issuecomment-706770430).

To test:

1. Open Discovery tab
2. Scroll to 'Sites to follow' card
3. Click follow/unfollow button
4. Verify site was followed/unfollowed

**Open Questions**

1. Now that we have a recommended blogs section in the man discover feed, should we remove "Sites you may like" section from "Manage Topics and Sites" and corresponding `ReaderRecommendedBlog` table?

TODO

1. ~~Need to clear recommended blogs on pull-to-refresh~~ [here](https://github.com/wordpress-mobile/WordPress-Android/blob/9e59a46e33ea0fe7e79b7d5eaa96bfd8a7866fcc/WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt#L216-L219) - Done in #13172
2. Add suggested site tracking (`reader_suggested_site_toggle_follow`, `reader_suggested_site_visited`) - Done in #13173

**Review/ Merge Instructions**

1. Review #13172,  #13173
2. Merge #13172,  #13173 to this PR
3. Test the app for all changes in this PR 
4. Merge this PR to develop

PR submission checklist:
